intellij-plugin-structure/tests/src/test/kotlin/com/jetbrains/plugin/structure/intellij/verifiers/StatusBarWidgetFactoryExtensionPointVerifierTest.kt

```diff
@@ -0,0 +1,66 @@
+package com.jetbrains.plugin.structure.intellij.verifiers
+
+import com.jetbrains.plugin.structure.base.plugin.PluginProblem
+import com.jetbrains.plugin.structure.intellij.plugin.IdePluginImpl
+import org.jdom2.Element
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+
+private const val PLUGIN_ID = "com.example.thirdparty"
+private const val PLUGIN_VENDOR = "PluginIndustries s.r.o."
+private const val EP_IMPLEMENTATION = "com.example.MyStatusBarWidgetFactory"
+
+private const val MESSAGE_TEMPLATE = "Extension Point in the <com.intellij.statusBarWidgetFactory> element " +
+  "must have 'id' attribute set for the [$EP_IMPLEMENTATION] implementation. " +
+  "Furthermore, it must match the value returned from the getId() method of the implementation."
+
+class StatusBarWidgetFactoryExtensionPointVerifierTest {
+  private lateinit var verifier: StatusBarWidgetFactoryExtensionPointVerifier
+
+  private lateinit var problems: MutableList<PluginProblem>
+
+  private val problemRegistrar = ProblemRegistrar {
+    problems += it
+  }
+
+  @Before
+  fun setUp() {
+    verifier = StatusBarWidgetFactoryExtensionPointVerifier()
+    problems = mutableListOf()
+  }
+
+
+  @Test
+  fun `status bar widget factory extension does not declare ID`() {
+    val epElement = Element("statusBarWidgetFactory").apply {
+      setAttribute("implementation", EP_IMPLEMENTATION)
+    }
+
+    val idePlugin = IdePluginImpl().apply {
+      pluginId = PLUGIN_ID
+      vendor = PLUGIN_VENDOR
+      extensions["com.intellij.statusBarWidgetFactory"] = mutableListOf(epElement)
+    }
+    verifier.verify(idePlugin, problemRegistrar)
+    Assert.assertEquals(1, problems.size)
+    val problem = problems[0]
+    Assert.assertEquals(MESSAGE_TEMPLATE.format("com.intellij.projectService"), problem.message)
```

Comment:
projectService?!

Reply:
Leftover from the previous test. Thanks for spotting this.